#7881 Inbound Shipment - round up to next whole packs

aimee-mcneil-melville opened the pull request:
<!-- IMPORTANT!
  - Every PR must reference an issue; this helps to explain the intent of the PR
 -->

Fixes #7881

# 👩🏻‍💻 What does this PR do?

<!-- Explain the changes you made -->

When entering stock in inbound shipments, packs are rounded up to the nearest whole pack.

This resolves decimal issues from the issue and we want to support only whole packs for an inbound shipment.

Shows an alert message to inform the user that the units have been rounded up.



https://github.com/user-attachments/assets/35135a12-33e7-44d6-ac00-60d04d8351b2




<!-- why are the changes needed -->

<!-- Add a screenshot if there are UI changes  -->

## 💌 Any notes for the reviewer?

I had some issues with NaN during development, looks to be resolved but something to keep an eye out for

Should the alert persist when adding another batch or changing other values eg packs received will also change the units received, but the alert no longer matches the inputs
<!-- Do you have any specific questions for the reviewer? -->

<!-- Is there a high risk/complicated change they should focus on? -->

<!-- any general areas of the codebase touched? any side effects caused? -->

<!-- Anything half cooked but going to be finished off in a different PR? -->

# 🧪 Testing

<!-- Explain the steps you'd take to test the changes of this PR manually -->

- [ ] Go to Inbound Shipments -> select a shipment or create one -> add a vaccine item to the shipment
- [ ] Enter a`Pack Size` of more than 1
- [ ] Change the `Units Received` received to a number that would not equal a whole number of packs (eg 8 when pack size is 5) -> see `Units Received` round up to the next whole pack
- [ ] See the alert appear detailing the difference between entered units and rounded up units
- [ ] Enter `Units Received` as a multiple of pack size -> see the alert disappear
- [ ] See that changing values in `Pack Size` and `Packs Received` results in correct calculations
- [ ] Check that when `Units Received` has rounded up, that the rounded value is saved

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: <!-- _(e.g.)_ New `issued` column in `Requisitions` indicates stock quantity already in shipments -->
  1.
  2.


# 📃 Reviewer Checklist

The PR Reviewer(s) should fill out this section before approving the PR

**Breaking Changes**
- [ ] No Breaking Changes in the Graphql API
- [ ] Technically some Breaking Changes but not expected to impact any integrations

**Issue Review**
- [ ] All requirements in original issue have been covered
- [ ] A follow up issue(s) have been created to cover additional requirements

**Tests Pass**
- [ ] Postgres
- [ ] SQLite
- [ ] Frontend

noel-yeldos reviewed (changes requested):
Thanks Aimee! It works great on first user input. 

Subsequent inputs don't seem to re-calculate the rounded value and so the alert persisting is confusing. See video for unexpected behaviour:

https://github.com/user-attachments/assets/d08c5aa8-9a85-4c06-a330-86afc726bd02

aimee-mcneil-melville commented:
Thanks Noel!
I've added some additional state updates to clear the message, and returned the rounded value to the UI so that it will 're-round' when needed.
I also added a longer debounce time, as if a user is trying to type (especially on a tablet) we don't want this rounding too fast for them to input the correct value

noel-yeldos commented on client/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx:
```diff
@@ -125,6 +136,7 @@ export const QuantityTableComponent = ({
       unit: pluralisedUnitName,
     }),
     width: 100,
+    cellProps: { debounce: 1000 },
```
Just from testing this feels a bit long - 500 seems smoother to me. What do you think?

noel-yeldos reviewed (changes requested):
Thanks for the changes Aimee! It's working smoothly now :) 

Just one thing that may still cause some confusion. When click 'add batch' the message still persists. What do you think about hiding the previous message when we click 'add batch'?

noel-yeldos commented on client/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/TabLayout.tsx:
```diff
@@ -97,7 +101,10 @@ export const TabLayout = ({
             variant="outlined"
-            onClick={addDraftLine}
+            onClick={() => {
+              addDraftLine;
```
@aimee-mcneil-melville just need to call the function here :) `addDraftLine();`

noel-yeldos reviewed (changes requested):
@aimee-mcneil-melville sorry just left a comment to call the `addDraftLines` function!

noel-yeldos reviewed (approved):
Thanks for the changes Aimee, LGTM! 🚀
